source/brailleDisplayDrivers/albatross/driver.py

```diff
@@ -0,0 +1,758 @@
+				pass  # enclosed ctrl packet
+			else:
+				log.debug(f"Partial ctrl packet {data}")
+				try:
+					while len(data) <= MAX_COMBINATION_KEYS:
+						oneKey = self._readQueue.popleft()
+						log.debug(f"Read: dequeued key {oneKey}, {len(self._readQueue)} items left")
+						# Encloses ctrl packet
+						if ord(oneKey) == data[0]:
+							# No reason but easier debugging to add oneKey to data
+							data += oneKey
+							break
+						data += oneKey
+				except IndexError:
+					self._waitingCtrlPacket = True
+					self._partialCtrlPacket = data
+					log.debug(
+						f"Read: Ctrl key packet {data} dequeued partially, "
+						"_readQueue is empty", exc_info=True
+					)
+					return
+				if len(data) > MAX_COMBINATION_KEYS and data[len(data) - 1] != data[0]:
+					self._waitingCtrlPacket = False
+					self._partialCtrlPacket = None
+					log.debug(f"Not valid key combination, ignoring {data}")
+					return
+		if self._keyLayout != KeyLayout.normal:
+			# Using custom key layout
+			data = self._changeKeyValues(
+				bytearray(data)
+			)
+		log.debug(f"Keys for key press: {data}")
+		pressedKeys = set(data)
+		log.debug(f"Forwarding keys {pressedKeys}")
+		try:
+			inputCore.manager.executeGesture(
+				gestures.InputGestureKeys(pressedKeys, self.name)
+			)
+		# Attribute error which rarely occurs here is something strange.
+		except (inputCore.NoInputGestureAction, AttributeError):
+			log.debug("", exc_info=True)
+		if self._waitingCtrlPacket:
+			self._waitingCtrlPacket = False
+			self._partialCtrlPacket = None
+
+	def _changeKeyValues(self, data: bytearray) -> bytes:
+		"""Changes pressed keys values according to current key layout.
+		@param data: pressed keys values.
+		@return: pressed keys values based on current key layout.
+		"""
+		for i, key in enumerate(data):
+			if self._keyLayout == KeyLayout.switched:
+				if key in LEFT_RIGHT_KEY_CODES.keys():
+					data[i] = LEFT_RIGHT_KEY_CODES[key]
+				elif key in LEFT_RIGHT_KEY_CODES.values():
+					j = list(
+						LEFT_RIGHT_KEY_CODES.values()
+					).index(key)
+					data[i] = list(
+						LEFT_RIGHT_KEY_CODES.keys()
+					)[j]
+				continue
+			if self._keyLayout == KeyLayout.bothSidesAsRight:
+				if key in LEFT_RIGHT_KEY_CODES.keys():
+					data[i] = LEFT_RIGHT_KEY_CODES[key]
+				continue
+			if self._keyLayout == KeyLayout.bothSidesAsLeft:
+				if key in LEFT_RIGHT_KEY_CODES.values():
+					j = list(
+						LEFT_RIGHT_KEY_CODES.values()
+					).index(key)
+					data[i] = list(
+						LEFT_RIGHT_KEY_CODES.keys()
+					)[j]
+		return bytes(data)
+
+	def _keepConnected(self):
+		"""Keep display connected if nothing is sent for a while."""
+		if self._tryToConnect or not self.numCells:
+			return
+		if time.time() - self._writeTime >= KC_INTERVAL:
+			self._writeQueue.append(BOTH_BYTES)
+			self._somethingToWrite()
+
+	def _clearOldCells(self):
+		"""Whole display should be updated at next time."""
+		for i in range(len(self._oldCells)):
+			self._oldCells[i] = 0

```

Comment:
can we just replace `_oldCells`?

```suggestion
		self._oldCells = [0] * len(self._oldCells)
```